dimension/client.go

```diff
@@ -100,37 +141,76 @@ func (c *Client) GetAreaTypes(ctx context.Context, userAuthToken, serviceAuthTok
 		}
 	}()
 
-	if resp.StatusCode == http.StatusNotFound {
-		var errorResp ErrorResp
-		if err := json.NewDecoder(resp.Body).Decode(&errorResp); err == nil {
-			return GetAreaTypesResponse{}, dperrors.New(
-				fmt.Errorf("error response from Dimensions API (%d): %w", resp.StatusCode, errorResp),
-				http.StatusInternalServerError,
-				logData,
-			)
-		}
+	if err := checkDimensionGetResponse(logData, *resp); err != nil {
+		return GetAreaTypesResponse{}, err
 	}
 
-	if resp.StatusCode != http.StatusOK {
-		// Best effort — an empty body is fine for the error message
-		body, _ := io.ReadAll(resp.Body)
+	var areaTypes GetAreaTypesResponse
+	if err := json.NewDecoder(resp.Body).Decode(&areaTypes); err != nil {
 		return GetAreaTypesResponse{}, dperrors.New(
-			fmt.Errorf("error response from Dimensions API (%d): %s", resp.StatusCode, body),
+			fmt.Errorf("unable to deserialize area types response: %w", err),
 			http.StatusInternalServerError,
 			logData,
 		)
 	}
 
-	var areaTypes GetAreaTypesResponse
-	if err := json.NewDecoder(resp.Body).Decode(&areaTypes); err != nil {
-		return GetAreaTypesResponse{}, dperrors.New(
-			fmt.Errorf("unable to deserialize area types response: %w", err),
+	return areaTypes, nil
+}
+
+func (c *Client) GetAreas(ctx context.Context, userAuthToken, serviceAuthToken, datasetID, areaTypeID, text string) (GetAreasResponse, error) {
+	logData := log.Data{
+		"method":       http.MethodGet,
+		"dataset_id":   datasetID,
+		"area_type_id": areaTypeID,
+		"text":         text,
+	}
+
+	urlPath := "areas"
+	urlValues := url.Values{"dataset": []string{datasetID}, "area-type": []string{areaTypeID}}
+	if text != "" {
+		urlValues.Add("text", text)
+	}
+
+	req, err := c.createDimensionGetRequest(ctx, userAuthToken, serviceAuthToken, urlPath, logData, urlValues)
+	if err != nil {
+		return GetAreasResponse{}, dperrors.New(
+			err,
+			http.StatusBadRequest,
+			logData,
+		)
+	}
```

Comment:
I _think_ by doing this you wipe out the status code provided by `createDimensionGetRequest` (i.e. a `500` for a garbage URL`). In order to return a status code from those, you'd need to probably pull the status code at the callsite?

```go
	req, err := c.createDimensionGetRequest(ctx, userAuthToken, serviceAuthToken, urlPath, logData, urlValues)
	if err != nil {
		return GetAreasResponse{}, dperrors.New(
			err,
			dperrors.StatusCode(err),
			logData,
		)
	}
```

Might be wrong but worth checking. If you do change it, then you probably don't need to pass `logData` down to `createDimensionGetRequest` either since it doesn't add anything to it.